packages/shylu/shylu_node/tacho/src/impl/Tacho_NumericTools_LevelSet.hpp

```diff
@@ -2719,15 +2725,17 @@ class NumericToolsLevelSet : public NumericToolsBase<ValueType, DeviceType> {
 #if (11000 > CUDA_VERSION)
       /// cuda 11.1 below
       const ordinal_type team_size_factor[2] = {32, 64}, vector_size_factor[2] = {8, 4};
+      const ordinal_type team_size_update[2] = {16, 8},  vector_size_update[2] = {16, 32};
 #else
       /// cuda 11.1 and higher
       const ordinal_type team_size_factor[2] = {64, 64}, vector_size_factor[2] = {8, 4};
+      const ordinal_type team_size_update[2] = {16, 8},  vector_size_update[2] = {32, 32};
 #endif
 #else
       /// not cuda
       const ordinal_type team_size_factor[2] = {64, 64}, vector_size_factor[2] = {8, 4};
+      const ordinal_type team_size_update[2] = {16, 8},  vector_size_update[2] = {32, 32};
```

Comment:
This would be safer for non-cuda devices where vector length 32 might be too many:
```
      cont int vmax = exec_space::vector_length_max();
      const ordinal_type team_size_update[2] = {16, 8},  vector_size_update[2] = {std::min(vmax, 32), std::min(vmax, 32)};
```
Could do the same for vector_size_factor as well.